presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -616,6 +618,26 @@ public PlanWithProperties visitFilter(FilterNode node, PreferredProperties prefe
             return rebaseAndDeriveProperties(node, planChild(node, preferredProperties));
         }
 
+        @Override
+        public PlanWithProperties visitSequence(SequenceNode node, PreferredProperties preferredProperties)
+        {
+            List<PlanWithProperties> leftPlans = node.getCteProducers().stream()
+                    .map(source -> accept(source, PreferredProperties.any()))
+                    .collect(ImmutableList.toImmutableList());
+            PlanWithProperties rightPlan = accept(node.getPrimarySource(), PreferredProperties.any());
+            List<PlanNode> childrenNodes = Stream.concat(
+                    leftPlans.stream().map(PlanWithProperties::getNode),
+                    Stream.of(rightPlan.getNode())
+            ).collect(ImmutableList.toImmutableList());
+            List<ActualProperties> actualPropertiesList = Stream.concat(
+                    leftPlans.stream().map(PlanWithProperties::getProperties),
+                    Stream.of(rightPlan.getProperties())
+            ).collect(ImmutableList.toImmutableList());
```

Comment:
The output property of the sequence node should be the property of the primary source (right plan) only?